lgc/builder/InOutBuilder.cpp

```diff
@@ -479,8 +479,33 @@ Value *InOutBuilder::modifyAuxInterpValue(Value *auxInterpValue, InOutInfo input
       else
         auxInterpValue = evalIjOffsetSmooth(auxInterpValue);
     }
-  } else
+  } else {
     assert(inputInfo.getInterpMode() == InOutInfo::InterpModeCustom);
+    if (inputInfo.isAdjustIj()) {
+      unsigned primitiveMode = getPipelineState()->getVerticesPerPrimitive();
+      switch (primitiveMode) {
+      case 1:
+        // points
+        break;
+      case 2:
+        // lines
+        break;
+      case 3: {
+        // triangles
+        // V0 ==> Attr_indx2
+        // V1 ==> Attr_indx0
+        // V2 ==> Attr_indx1
+        // just satisfies (idx + 2)%3.
```

Comment:
I can't understand the algorithm, could you explain more about it?

Reply:
The input variable decorated by 'PerVertexKHR' is an array which length is 3, but array[0] matches vertex2, array[1] matches vertex0, array[2] matches vertex1, so here is to adjust correspondence to make array[0] => vertex0, array[1] = vertex1...